src/baconian.rs

```diff
@@ -0,0 +1,467 @@
+        "Z" => '\u{1D44d}',
+        // Using Mathematical Sans-Serif Italic
+        "a" => '\u{1D622}',
+        "b" => '\u{1D623}',
+        "c" => '\u{1D624}',
+        "d" => '\u{1D625}',
+        "e" => '\u{1D626}',
+        "f" => '\u{1D627}',
+        "g" => '\u{1D628}',
+        "h" => '\u{1D629}',
+        "i" => '\u{1D62a}',
+        "j" => '\u{1D62b}',
+        "k" => '\u{1D62c}',
+        "l" => '\u{1D62d}',
+        "m" => '\u{1D62e}',
+        "n" => '\u{1D62f}',
+        "o" => '\u{1D630}',
+        "p" => '\u{1D631}',
+        "q" => '\u{1D632}',
+        "r" => '\u{1D633}',
+        "s" => '\u{1D634}',
+        "t" => '\u{1D635}',
+        "u" => '\u{1D636}',
+        "v" => '\u{1D637}',
+        "w" => '\u{1D638}',
+        "x" => '\u{1D639}',
+        "y" => '\u{1D63a}',
+        "z" => '\u{1D63b}'
+    };
+}
+
+/// Get the code for a given key (source character)
+fn get_code(distinct: bool, key: &str) -> String {
+    let mut code = String::new();
+    if distinct {
+        if DISTINCT_CODES.contains_key(key.to_uppercase().as_str()) {
+            code.push_str(DISTINCT_CODES.get(key.to_uppercase().as_str()).unwrap());
+        }
+    } else {
+        // Need to handle 'I'/'J' and 'U'/'V'
+        let mut key_upper = key.to_uppercase();
+
+        match key_upper.as_str() {
+            "J" => key_upper = "I".to_string(),
+            "U" => key_upper = "V".to_string(),
+            _ => {}
+        }
+        if TRAD_CODES.contains_key(key_upper.as_str()) {
+            code.push_str(TRAD_CODES.get(key_upper.as_str()).unwrap());
+        }
+    }
+
+    code
+}
+
+/// Gets the key (the source character) for a given cipher text code
+fn get_key(distinct: bool, code: &str) -> String {
+    let mut key = String::new();
+
+    let codes = if distinct {
+        DISTINCT_CODES.iter()
+    } else {
+        TRAD_CODES.iter()
+    };
+    for (_key, val) in codes {
+        if val == &code {
+            key.push_str(_key);
+        }
+    }
+    key
+}
+
+/// This struct is created by the `new()` method. See its documentation for more.
+pub struct Baconian {
+    distinct: bool,
+    decoy_text: String,
+}
+
+impl Cipher for Baconian {
+    type Key = (bool, Option<String>);
+    type Algorithm = Baconian;
+
+    /// Initialise a Baconian cipher
+    ///
+    /// The `key` tuple maps to the following:
+    ///     `(bool, Option<str>) =
+    ///         (distinct, decoy_text)`.
+    ///
+    /// Where ...
+    ///
+    /// * The encoding will be distinct for all alphabetical characters, or classical
+    ///     where I, J, U and V are mapped to the same value pairs
+    /// * An optional decoy message that will will be used to hide the message -
+    ///     default is boilerplate "Lorem ipsum" text.
+    ///
+    fn new(key: (bool, Option<String>)) -> Result<Baconian, &'static str> {
+        Ok(Baconian {
+            distinct: key.0,
+            decoy_text: key.1.unwrap_or_else(|| String::from(DEFAULT_DECOY)),
```

Comment:
Rather using a static default decoy text, perhaps we could randomly generate a string as the default decoy? I know this might diverge from other implementations but I think it would be a cool feature.

Reply:
I need to look into that. Easy thing in Python, not sure about Rust, maybe there is a crate that could be used. I agree, it would be cool, but I'll park this for now. Maybe make an issue for it..?

Reply:
It might be not that hard to integrate, see the crate: https://github.com/mgeisler/lipsum. However I understand if you want to leave this out for now, we can create an issue.

Reply:
Will have a look. Will flag whether it would be better as an issue. It would certainly be interesting, and allow the default always to manage the size of the plaintext